test/lib/y2users/user_test.rb

```diff
@@ -0,0 +1,451 @@
+
+            expect(group).to be_a(Y2Users::Group)
+            expect(group.gid).to eq(100)
+          end
+        end
+      end
+    end
+  end
+
+  describe "#groups" do
+    let(:group1) do
+      group = Y2Users::Group.new("test1")
+      group.gid = 100
+      group.users_name = []
+      group
+    end
+
+    let(:group2) do
+      group = Y2Users::Group.new("test2")
+      group.gid = 101
+      group.users_name = ["test", "other"]
+      group
+    end
+
+    let(:group3) do
+      group = Y2Users::Group.new("test3")
+      group.gid = 102
+      group.users_name = ["test"]
+      group
+    end
+
+    let(:group4) do
+      group = Y2Users::Group.new("test4")
+      group.gid = 103
+      group.users_name = ["other"]
+      group
+    end
+
+    let(:config) { Y2Users::Config.new }
+
+    let(:groups) { [group1, group2, group3, group4] }
+
+    before do
+      config.attach(groups)
+    end
+
+    context "if the user is not attached to any config" do
+      it "returns an empty list" do
+        expect(subject.groups).to be_empty
+      end
+    end
+
+    context "if the user is attached to a config" do
+      before do
+        config.attach(subject)
+      end
+
+      it "includes all groups from config that contain such a user name" do
+        expect(subject.groups).to contain_exactly(group2, group3)
+      end
+
+      context "and the config has no groups with such a user name" do
+        let(:groups) { [group1, group4] }
+
+        it "returns an empty list" do
+          expect(subject.groups).to be_empty
+        end
+      end
+
+      context "and the user indicates the gid of its primary group" do
+        before do
+          subject.gid = 100
+        end
+
+        context "and the config has a group with such a gid" do
+          it "includes the primary group" do
+            expect(subject.groups).to include(group1)
+          end
+        end
+
+        context "and the config has not a group with such a gid" do
+          let(:groups) { [group2, group3, group4] }
+
+          it "does not include the primary group" do
+            expect(subject.groups).to_not include(group1)
+          end
+        end
+      end
+    end
+  end
+
+  describe "#expire_date" do
+    before do
+      subject.password = password
+    end
+
+    context "when the user has no password" do
+      let(:password) { nil }
+
+      it "returns nil" do
+        expect(subject.expire_date).to be_nil
+      end
+    end
+
+    context "when the user has a password" do
+      let(:password) { Y2Users::Password.create_plain("S3cr3T") }
+
+      before do
+        password.account_expiration = expiration
+      end
+
+      context "and the password has no expiration" do
+        let(:expiration) { nil }
+
+        it "returns nil" do
+          expect(subject.expire_date).to be_nil
+        end
+      end
+
+      context "and the password has expiration" do
+        let(:expiration) { today }
+
+        let(:today) { Date.today }
```

Comment:
NP: this can be done in the same line

Reply:
Only to not repeat `Date.today` below in the check.